flatgfa-py/src/lib.rs

```diff
@@ -126,6 +127,97 @@ impl PyFlatGFA {
         mmap.flush()?;
         Ok(())
     }
+    
+    #[getter]
+    fn size(&self) -> usize {
+        let gfa = self.0.view();
+        file::size(&gfa)
+    }
+
+    fn print_gaf_lookup(&self, gaf: &str) {
+        let gfa = self.0.view();
+
+        let name_map = flatgfa::namemap::NameMap::build(&gfa);
+
+        let gaf_buf = flatgfa::memfile::map_file(&gaf);
+        let parser = flatgfa::ops::gaf::GAFParser::new(&gaf_buf);
+
+        // Print the actual sequences for each chunk in the GAF.
+        for read in parser {
+            print!("{}\t", read.name);
+            for event in flatgfa::ops::gaf::PathChunker::new(&gfa, &name_map, read) {
+                event.print_seq(&gfa);
+            }
+            println!();
+        }
+    }
+
+    fn test_gaf(&self, gaf: &str) -> Vec<Vec<PyChunkEvent>> {
+        let gfa = self.0.view();
+
+        let name_map = flatgfa::namemap::NameMap::build(&gfa);
+
+        let gaf_buf = flatgfa::memfile::map_file(&gaf);
+        let parser = flatgfa::ops::gaf::GAFParser::new(&gaf_buf);
+
+        let r: Vec<Vec<PyChunkEvent>> = parser.into_iter().map(
+                |x| flatgfa::ops::gaf::PathChunker::new(&gfa, &name_map, x)
+                    .into_iter()
+                    .map(|c| PyChunkEvent {
+                        chunk_event: c.into(),
+                        gfa: self.0.clone(),
+                    })
+                    .collect()
+            ).collect();
+            
+        r
+    }
+}
+
+#[pyclass(frozen)]
+#[pyo3(name = "ChunkEvent", module = "flatgfa")]
+struct PyChunkEvent {
+    chunk_event: Arc<ChunkEvent>,
+    gfa: Arc<Store>,
+}
+
+#[pymethods]
+impl PyChunkEvent {
+    fn __str__(&self) -> String {
+        "".to_string()
+    }
+
+    #[getter]
+    fn handle(&self) -> String {
+        let seg_num: u32 = self.chunk_event.handle.segment().into();
+        format!("{}, {}", &self.chunk_event.handle.orient(), seg_num)
+    }
+
+    #[getter]
+    fn range(&self) -> String {
```

Comment:
Similarly, this could return a pair of Python integers instead of the stringified version.

Reply:
I wasn't quite sure which convention to use for the `None` and `All` case. I assume the range is inclusive on both ends, so I made `All` return `(0, size - 1)` while `None` returns `(1, 0)`, but I can change that if we want.